tidy3d/plugins/smatrix/ports/rectangular_lumped.py

```diff
@@ -296,3 +297,204 @@ def _to_current_box(self, grid: Grid) -> Box:
+        # get terminals' bounds
+        ground_bounds = np.array(ground_2d.bounds).reshape(2, 2).T
+        signal_bounds = np.array(signal_2d.bounds).reshape(2, 2).T
+
+        def intervals_overlap(a, b):
+            """Return True if [a_min, a_max] and [b_min, b_max] overlap."""
+            a_min, a_max = a
+            b_min, b_max = b
+            return not (a_max < b_min or a_min > b_max)
+
+        # ensure that terminal bounding boxes don't overlap along voltage axis
+        if intervals_overlap(ground_bounds[voltage_axis_2d], signal_bounds[voltage_axis_2d]):
+            raise ValidationError(
+                "Ground and signal terminals have overlapping bounds along voltage axis."
+            )
+
+        if not intervals_overlap(ground_bounds[lateral_axis_2d], signal_bounds[lateral_axis_2d]):
+            raise ValidationError(
+                "Ground and signal terminals don't have overlapping bounds along lateral axis."
+            )
+
+        # reshape and combine bounds of the signal and ground cuts; rows correspond to axes
+        bounds = np.hstack((ground_bounds, signal_bounds))
+
+        # for each axis extract port bounds
+        new_bounds = np.sort(bounds, axis=1)[:, 1:3]
+        vmin, vmax = new_bounds[voltage_axis_2d, :]
+        lmin, lmax = new_bounds[lateral_axis_2d, :]
+
+        if port_width is not None:
+            if lateral_coord is not None:
+                # If port_width and lat_coord are specified, set based on those instead
+                lmin_new, lmax_new = (
+                    lateral_coord - port_width / 2,
+                    lateral_coord + port_width / 2,
+                )
+            else:
+                # If only port_width is specified, set based on bounds
+                lcenter = (lmin + lmax) / 2
+                lmin_new, lmax_new = (lcenter - port_width / 2, lcenter + port_width / 2)
+                # make sure that port_width does not exceed width of terminal overlap along lateral axis.
+
+            if lmin_new < lmin or lmax_new > lmax:
+                raise ValueError(
+                    "Specified port region extends beyond the overlap between the signal and ground terminals."
+                )
+            # update port limits along lateral axis
+            lmin, lmax = lmin_new, lmax_new
+
+        # WARNING: This currently does not check if edges of lumped port touches ground/signal.
+        #          This can happen if ground/signal does not overlap along lateral_axis
+        #          It is up to the user to provide sensible geometry choices for ground/signal to avoid this
+
+        # allocate memory for lumped port bounds
+        rmin = np.zeros(3)
+        rmax = np.zeros(3)
+
+        # construct lumped port bounds
+        for axis, bounds in (
+            (voltage_axis, (vmin, vmax)),
+            (lateral_axis, (lmin, lmax)),
+            (normal_axis, (normal_coord, normal_coord)),
+        ):
+            rmin[axis], rmax[axis] = bounds
```

Comment:
probably this for loop doesn't add any convenience or clarity, can just do something simple:
```
rmin[voltage_axis], rmax[voltage_axis] = (vmin, vmax)
rmin[lateral_axis], rmax[lateral_axis] = (lmin, lmax)
rmin[normal_axis], rmax[normal_axis] = (normal_coord, normal_coord)
```
or
```
axes = [voltage_axis, lateral_axis, normal_axis]
rmin[axes] = (vmin, lmin, normal_coord)
rmax[axes] = (vmax, lmax, normal_coord)
```